presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java

```diff
@@ -657,4 +651,30 @@ public SavedRow(Page page, int position, long joinPositionWithinPartition, boole
             this.joinSourcePositions = joinSourcePositions;
         }
     }
+
+    private boolean tryBuildPage()
+    {
+        if (pageBuilder.isEmpty()) {
+            return false;
+        }
+
+        if (finished || pageBuilder.isFull()) {
```

Comment:
Not sure why you consult `finished` here?

BTW this seems to be always `false` when this method is called. However, if you consulted `finishing` it wouldn't be good either (you would probably produce single-row pages for whole last page of the input).

Reply:
I don't think we need need to check `finished` or `finishing` actually. Given we will always produce a page at the end of a probe, there is no way we can end up a state where we have to force flushing a page when the operator has no more input.